app/src/main/java/de/danoeh/antennapod/config/StorageCallbacksImpl.java

```diff
@@ -149,5 +150,41 @@ public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
             db.execSQL(PodDBAdapter.CREATE_INDEX_QUEUE_FEEDITEM);
             db.execSQL(PodDBAdapter.CREATE_INDEX_SIMPLECHAPTERS_FEEDITEM);
         }
+        if(oldVersion <= 15) {
+            db.execSQL("ALTER TABLE " + PodDBAdapter.TABLE_NAME_FEED_MEDIA
+                    + " ADD COLUMN " + PodDBAdapter.KEY_HAS_EMBEDDED_PICTURE + " INTEGER DEFAULT -1");
+            Thread t = new Thread(new Runnable() {
+                public void run() {
+                    db.execSQL("UPDATE " + PodDBAdapter.TABLE_NAME_FEED_MEDIA
+                            + " SET " + PodDBAdapter.KEY_HAS_EMBEDDED_PICTURE + "=0"
+                            + " WHERE " + PodDBAdapter.KEY_DOWNLOADED + "=0");
+                    Cursor c = db.rawQuery("SELECT " + PodDBAdapter.KEY_FILE_URL + " FROM " + PodDBAdapter.TABLE_NAME_FEED_MEDIA +
```

Comment:
cursor is never closed (though probably not a big deal since this should only ever be run once)